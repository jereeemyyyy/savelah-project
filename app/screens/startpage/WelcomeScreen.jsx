import { SafeAreaView, Text, TouchableOpacity, View } from "react-native";
import { useNavigation } from "@react-navigation/native";
import React from "react";

export default function WelcomeScreen() {

    const navigation = useNavigation();

    return (
        <SafeAreaView className="bg-gray-800 flex-1" >
            <View className="flex-1 flex justify-around my-4">
                <View>
                    <Text className="text-white fond-bold text-center">
                        "Insert Some Image Here"
                    </Text>
                </View>
                <Text className="font-bold text-white text-4xl px-5 ">
                    Easiest way to manage your finances!
                </Text>
                <View>
                    <TouchableOpacity
                        onPress={() => navigation.navigate('SignUp')}
                        className="py-3 bg-gray-500 mx-7 rounded-xl">
                        <Text className="text-white font-bold text-center text-xl">
                            Get started
                        </Text>
                    </TouchableOpacity>
                    <View className="flex-row justify-center">
                        <Text className="text-white font-semibold py-3">
                            Already have an account?
                        </Text>
                        <TouchableOpacity onPress={() => navigation.navigate('Login')}>
                            <Text className="text-indigo-500 font-semibold py-3 px-2">
                                Login
                            </Text>
                        </TouchableOpacity>
                    </View>
                </View>
            </View>

        </SafeAreaView>
    );
}