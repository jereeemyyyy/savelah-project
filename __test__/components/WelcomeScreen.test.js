import React from 'react';
import renderer from 'react-test-renderer';
import WelcomeScreen from '../../app/screens/startpage/WelcomeScreen'; // Update with the correct path

jest.mock('@react-navigation/native', () => {
    const actualNav = jest.requireActual('@react-navigation/native');
    return {
        ...actualNav,
        useNavigation: () => ({
            navigate: jest.fn(),
        }),
    };
});

describe('WelcomeScreen', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<WelcomeScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains the expected text elements', () => {
        const component = renderer.create(<WelcomeScreen />);
        const instance = component.root;

        expect(instance.findByProps({ testID: 'headerText' }).props.children).toBe('Easiest way to manage your finances!');
        expect(instance.findByProps({ testID: 'getStartedButtonText' }).props.children).toBe('Get started');
        expect(instance.findByProps({ testID: 'loginPromptText' }).props.children).toBe('Already have an account?');
        expect(instance.findByProps({ testID: 'loginButtonText' }).props.children).toBe('Login');
    });

    it('navigates to SignUp screen when Get started button is pressed', () => {
        const mockNavigate = jest.fn();

        jest.spyOn(require('@react-navigation/native'), 'useNavigation').mockReturnValue({ navigate: mockNavigate });

        const component = renderer.create(<WelcomeScreen />);
        const instance = component.root;

        const getStartedButton = instance.findByProps({ testID: 'getStartedButton' });
        getStartedButton.props.onPress();

        expect(mockNavigate).toHaveBeenCalledWith('SignUp');
    });

    it('navigates to Login screen when Login button is pressed', () => {
        const mockNavigate = jest.fn();

        jest.spyOn(require('@react-navigation/native'), 'useNavigation').mockReturnValue({ navigate: mockNavigate });

        const component = renderer.create(<WelcomeScreen />);
        const instance = component.root;

        const loginButton = instance.findByProps({ testID: 'loginButton' });
        loginButton.props.onPress();

        expect(mockNavigate).toHaveBeenCalledWith('Login');
    });
});
