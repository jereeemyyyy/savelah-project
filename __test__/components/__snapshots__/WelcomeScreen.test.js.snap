// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen renders correctly 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "backgroundColor": "#1f2937",
      },
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
  testID="welcomeScreen"
>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "display": "flex",
        },
        {
          "justifyContent": "space-around",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
          },
        ]
      }
      testID="imageContainer"
    >
      <Image
        source={1}
        style={
          {
            "height": 200,
            "width": 350,
          }
        }
        testID="welcomeImage"
      />
    </View>
    <View
      style={
        [
          {
            "marginTop": -80,
          },
        ]
      }
      testID="headerTextContainer"
    >
      <Text
        style={
          [
            {
              "fontWeight": "700",
            },
            {
              "color": "#fff",
            },
            {
              "fontSize": 36,
              "lineHeight": 40,
            },
            {
              "paddingLeft": 20,
              "paddingRight": 20,
            },
          ]
        }
        testID="headerText"
      >
        Easiest way to manage your finances!
      </Text>
    </View>
    <View
      testID="buttonsContainer"
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#6b7280",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "marginLeft": 28,
            "marginRight": 28,
            "opacity": 1,
            "paddingBottom": 12,
            "paddingTop": 12,
          }
        }
        testID="getStartedButton"
      >
        <Text
          style={
            [
              {
                "color": "#fff",
              },
              {
                "fontWeight": "700",
              },
              {
                "textAlign": "center",
              },
              {
                "fontSize": 20,
                "lineHeight": 28,
              },
            ]
          }
          testID="getStartedButtonText"
        >
          Get started
        </Text>
      </View>
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "justifyContent": "center",
            },
          ]
        }
        testID="loginContainer"
      >
        <Text
          style={
            [
              {
                "color": "#fff",
              },
              {
                "fontWeight": "600",
              },
              {
                "paddingBottom": 12,
                "paddingTop": 12,
              },
            ]
          }
          testID="loginPromptText"
        >
          Already have an account?
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="loginButton"
        >
          <Text
            style={
              [
                {
                  "color": "#6366f1",
                },
                {
                  "fontWeight": "600",
                },
                {
                  "paddingBottom": 12,
                  "paddingTop": 12,
                },
                {
                  "paddingLeft": 8,
                  "paddingRight": 8,
                },
              ]
            }
            testID="loginButtonText"
          >
            Login
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
